cpp/tests/experimental/renumbering_test.cpp

```diff
@@ -0,0 +1,181 @@
+// enabled by command line parameter s'--perf'
+//
+static int PERF = 0;
+
+struct Renumbering_Usecase {
+  bool check_correctness{true};
+};
+
+template <typename input_usecase_t>
+class Tests_Renumbering
+  : public ::testing::TestWithParam<std::tuple<Renumbering_Usecase, input_usecase_t>> {
+ public:
+  Tests_Renumbering() {}
+  static void SetupTestCase() {}
+  static void TearDownTestCase() {}
+
+  virtual void SetUp() {}
+  virtual void TearDown() {}
+
+  template <typename vertex_t, typename edge_t, typename weight_t>
+  void run_current_test(Renumbering_Usecase const& renumbering_usecase,
+                        input_usecase_t const& input_usecase)
+  {
+    raft::handle_t handle{};
+    HighResClock hr_clock{};
+
+    std::vector<vertex_t> h_original_src_v{};
+    std::vector<vertex_t> h_original_dst_v{};
+    std::vector<vertex_t> h_final_src_v{};
+    std::vector<vertex_t> h_final_dst_v{};
+
+    rmm::device_uvector<vertex_t> src_v(0, handle.get_stream_view());
+    rmm::device_uvector<vertex_t> dst_v(0, handle.get_stream_view());
+    rmm::device_uvector<vertex_t> renumber_map_labels_v(0, handle.get_stream_view());
+    vertex_t number_of_vertices{};
+    edge_t number_of_edges{};
+    bool symmetric{};
+
+    std::tie(src_v, dst_v, std::ignore, number_of_vertices, symmetric) =
+      input_usecase.template construct_edgelist<vertex_t, edge_t, weight_t, false, false>(handle,
+                                                                                          false);
+
+    if (renumbering_usecase.check_correctness) {
+      h_original_src_v.resize(src_v.size());
+      h_original_dst_v.resize(dst_v.size());
+
+      raft::update_host(h_original_src_v.data(), src_v.data(), src_v.size(), handle.get_stream());
+      raft::update_host(h_original_dst_v.data(), dst_v.data(), dst_v.size(), handle.get_stream());
+    }
+
+    if (PERF) {
+      handle.get_stream_view().synchronize();  // for consistent performance measurement
+      hr_clock.start();
+    }
+
+    std::tie(renumber_map_labels_v, std::ignore) =
+      cugraph::experimental::renumber_edgelist<vertex_t, edge_t, false>(
+        handle, std::nullopt, src_v.begin(), dst_v.begin(), src_v.size());
+
+    if (PERF) {
+      handle.get_stream_view().synchronize();  // for consistent performance measurement
+      double elapsed_time{0.0};
+      hr_clock.stop(&elapsed_time);
+      std::cout << "renumbering took " << elapsed_time * 1e-6 << " s.\n";
+    }
+
+    if (renumbering_usecase.check_correctness) {
+      cugraph::experimental::unrenumber_local_int_vertices(
+        handle,
+        src_v.data(),
+        src_v.size(),
+        renumber_map_labels_v.data(),
+        0,
+        static_cast<vertex_t>(renumber_map_labels_v.size()));
+      cugraph::experimental::unrenumber_local_int_vertices(
+        handle,
+        dst_v.data(),
+        dst_v.size(),
+        renumber_map_labels_v.data(),
+        0,
+        static_cast<vertex_t>(renumber_map_labels_v.size()));
```

Comment:
This doesn't matter in SG, but if we consider MG extension, ```unrenumber_local_int_vertices``` works only with vertices in the range of ```[graph_view.get_local_vertex_first(), graph_view.get_local_vertex_last())```; so unrenumber_int_vertices (https://github.com/rapidsai/cugraph/blob/branch-21.10/cpp/include/cugraph/experimental/graph_functions.hpp#L212) could be more extensible.

Also note the comment in https://github.com/rapidsai/cugraph/blob/branch-21.10/cpp/include/cugraph/experimental/graph_functions.hpp#L185. If we need to unrenumber local edges in the performance critical path, we need to add addition unrenumber functions for better performance.